route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -326,25 +281,6 @@ namespace route_following_plugin
         return true;
     }
 
-    void RouteFollowingPlugin::updateCurrentStatus(cav_msgs::Maneuver maneuver, double& speed, double& current_progress, int& lane_id){
-        if(maneuver.type == cav_msgs::Maneuver::STOP_AND_WAIT){
-            speed =0.0;
-            current_progress = maneuver.stop_and_wait_maneuver.end_dist;
-            lane_id = stoi(maneuver.stop_and_wait_maneuver.ending_lane_id);
-        }
-        else if(maneuver.type == cav_msgs::Maneuver::LANE_FOLLOWING){
-            speed =  maneuver.lane_following_maneuver.end_speed;
-            current_progress =  maneuver.lane_following_maneuver.end_dist;
-            lane_id =  stoi(maneuver.lane_following_maneuver.lane_id);
-        }
-        else{
-            speed = GET_MANEUVER_PROPERTY(maneuver,end_speed);
-            current_progress =GET_MANEUVER_PROPERTY(maneuver,end_dist);
-            lane_id = stoi(GET_MANEUVER_PROPERTY(maneuver,ending_lane_id));
-        }
-
-    }
-
```

Comment:
The findSpeedLimit method below can be substantially simplified. 
1. Get the traffic_rules optional
2. Check if it is valid. If not throw an exception as something has gone very very wrong
3. If the traffic rules object does exist all you need to do is call speedLimit(llt); Internally the carma_wm handles the config_limit and hard coded max speed limit constraints. You do not need to check them yourself